pkg/proxy/endpoints.go

```diff
@@ -132,15 +137,24 @@ func (ect *EndpointChangeTracker) Update(previous, current *v1.Endpoints) bool {
 		change = &endpointsChange{}
 		change.previous = ect.endpointsToEndpointsMap(previous)
 		ect.items[namespacedName] = change
+		if t := getLastChangeTriggerTime(endpoints); !t.IsZero() {
+			ect.lastChangeTriggerTimes[namespacedName] = t
+		}
 	}
 	change.current = ect.endpointsToEndpointsMap(current)
 	// if change.previous equal to change.current, it means no change
 	if reflect.DeepEqual(change.previous, change.current) {
 		delete(ect.items, namespacedName)
+		delete(ect.lastChangeTriggerTimes, namespacedName)
 	}
 	return len(ect.items) > 0
 }
 
+func getLastChangeTriggerTime(endpoints *v1.Endpoints) time.Time {
+	val, _ := time.Parse(time.RFC3339Nano, endpoints.Annotations[v1.EndpointsLastChangeTriggerTime])
```

Comment:
Don't silently ignore errors - if not more, at least log an error.

Reply:
Done. Added log statement and a comment explaining why we can ignore the error.